include/boost/geometry/formulas/differential_quantities.hpp

```diff
@@ -0,0 +1,269 @@
+// Boost.Geometry
+
+// Copyright (c) 2016 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Adam Wulkiewicz, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_INVERSE_DIFFERENTIAL_QUANTITIES_HPP
+#define BOOST_GEOMETRY_FORMULAS_INVERSE_DIFFERENTIAL_QUANTITIES_HPP
+
+
+#include <boost/geometry/util/condition.hpp>
+#include <boost/geometry/util/math.hpp>
+
+
+namespace boost { namespace geometry { namespace formula
+{
+
+/*!
+\brief The solution of a part of the inverse problem - differential quantities.
+\author See
+- Charles F.F Karney, Algorithms for geodesics, 2011
+https://arxiv.org/pdf/1109.4448.pdf
+*/
+template <
+    typename CT,
+    bool EnableReducedLength,
+    bool EnableGeodesicScale,
+    unsigned Order = 2,
+    bool ApproxF = true
+>
+class differential_quantities
+{
+public:
+    static inline void apply(CT const& lon1, CT const& lat1,
+                             CT const& lon2, CT const& lat2,
+                             CT const& azimuth, CT const& reverse_azimuth,
+                             CT const& b, CT const& f,
+                             CT & reduced_length, CT & geodesic_scale)
+    {
+        CT const dlon = lon2 - lon1;
+        CT const sin_lat1 = sin(lat1);
+        CT const cos_lat1 = cos(lat1);
+        CT const sin_lat2 = sin(lat2);
+        CT const cos_lat2 = cos(lat2);
+
+        apply(dlon, sin_lat1, cos_lat1, sin_lat2, cos_lat2,
+              azimuth, reverse_azimuth,
+              b, f,
+              reduced_length, geodesic_scale);
+    }
+
+    static inline void apply(CT const& dlon,
+                             CT const& sin_lat1, CT const& cos_lat1,
+                             CT const& sin_lat2, CT const& cos_lat2,
+                             CT const& azimuth, CT const& reverse_azimuth,
+                             CT const& b, CT const& f,
+                             CT & reduced_length, CT & geodesic_scale)
+    {
+        CT const one_minus_f = 1 - f;
+        CT const sin_bet1 = one_minus_f * sin_lat1;
+        CT const sin_bet2 = one_minus_f * sin_lat2;
```

Comment:
`1-f = 0 <=> f=1 <=> b=0` so it's a disk ("flat" ellipsoid). And `b` may be passed by the user so maybe it should be an exception. But since it's such unreasonable scenario it shouldn't be checked here each time a formula is used because in the vast majority of cases `f` will be reasonable. Instead it could be checked in `srs::spheroid`. What do you think?